org.code.toboggan.core/src/org/code/toboggan/core/api/file/FileCreate.java

```diff
@@ -0,0 +1,50 @@
+package org.code.toboggan.core.api.file;
+
+import java.nio.file.Path;
+
+import org.code.toboggan.core.api.AbstractAPICall;
+import org.code.toboggan.core.extension.ExtensionIDs;
+import org.code.toboggan.core.extension.ExtensionManager;
+import org.code.toboggan.core.extension.ICoreAPIExtension;
+import org.code.toboggan.core.extension.file.IFileCreateExtension;
+
+public class FileCreate extends AbstractAPICall {
+
+	private String name;
+	private Path workspaceRelativePath;
+	private long projectID;
+	private byte[] fileBytes;
+	
+	public FileCreate(ExtensionManager manager, String name, Path workspaceRelativePath, long projectID, byte[] fileBytes) {
+		this.extensions = manager.getExtensions(ExtensionIDs.FILE_CREATE_ID);
+		this.name = name;
+		this.workspaceRelativePath = workspaceRelativePath;
+		this.projectID = projectID;
+		this.fileBytes = fileBytes;
+	}
+
+	@Override
+	public void execute() {
+		for (ICoreAPIExtension e : this.extensions) {
+			IFileCreateExtension pExt = (IFileCreateExtension) e;
+			pExt.fileCreate(name, workspaceRelativePath, projectID, fileBytes);
```

Comment:
All the other functions are in past tense, so this should be `pExt.fileCreated` (refactor it in the extension as well)